code/game/area/areas_power.dm

```diff
@@ -0,0 +1,33 @@
+///**ALL POWER RELATED PROCS AND CODE DEFINED ON THE /area LEVEL**///
+
+/**
+  * Generate a power alert for this area
+  *
+  * Sends to all ai players, alert consoles, drones and alarm monitor programs in the world
+  */
+/area/proc/poweralert(state, obj/source)
+	if(state == poweralm)
+		return
+	poweralm = state
+	if(!istype(source))	//Only report power alarms on the z-level where the source is located.
+		return
```

Comment:
what does this have to do with Zlevles?

Reply:
cleaned up in next commit, this is probably just leftover stuff